rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/ArtifactEndpoint.java

```diff
@@ -113,6 +115,43 @@
             @Parameter(description = FILTER_MD5_DESC) @QueryParam("md5") String md5,
             @Parameter(description = FILTER_SHA1_DESC) @QueryParam("sha1") String sha1);
 
+    static final String GET_ALL_FILTERED_DESC = "Gets all artifacts according to specified filters.";
+    static final String FILTER_IDENTIFIER_DESC = "Filter by artifact identifier or its part.";
+    static final String FILTER_QUALITY_DESC = "List of artifact qualities to include in result.";
+    static final String FILTER_REPOSITORY_TYPE_DESC = "Type of target repository.";
+
+    /**
+     * {@value GET_ALL_FILTERED_DESC}
+     *
+     * @param pageParameters
+     * @param identifier {@value FILTER_IDENTIFIER_DESC}
+     * @param qualities {@value FILTER_QUALITY_DESC}
+     * @param repoType {@value FILTER_REPOSITORY_TYPE_DESC}
+     * @return
+     */
+    @Operation(
+            summary = GET_ALL_FILTERED_DESC,
+            responses = {
+                    @ApiResponse(
+                            responseCode = SUCCESS_CODE,
+                            description = SUCCESS_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ArtifactPage.class))),
+                    @ApiResponse(
+                            responseCode = INVALID_CODE,
+                            description = INVALID_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
+                    @ApiResponse(
+                            responseCode = SERVER_ERROR_CODE,
+                            description = SERVER_ERROR_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))) })
+    @GET
+    @Path("/filter")
+    Page<Artifact> getAllFiltered(
```

Comment:
What I don't get is why won't we just use RSQL on getArtifacts endpoint instead of creating entire new endpoint. DA needs to prepare request anyway.

RSQL that should be equivalent example is: q=identifier=like=group:*:version;artifactQuality=in=(DELETED,BLACKLISTED,NEW...);targetRepository.repositoryType==MAVEN
 
If you don't want to use =in= you can group multiple ORs like this: q=identifier=like=group:*:version;(artifactQuality==NEW,artifactQuality==TEMPORARY...);targetRepository.repositoryType==NPM 

RSQL will create just one query to DB.

Reply:
I have other question. Is it necessary for the endpoint to return whole Artifact? Isn't just subset of the data needed? For example, does DA need Build or TargetRepository?
I belive this endpoint should be really optimized.

Reply:
@michalovjan  I believe that the new endpoint was suggested because of optimization as Petr describes in his doc https://docs.engineering.redhat.com/display/JP/Design%3A+Limit+usage+of+Brew+pull -> (to avoid frequent rsql parsing). @pkocandr is that right? 

@janinko I am afraid I do not know what exactly DA needs from the Artifact. I can discuss this with Petr in more detail to crop the unnecessary data.

Reply:
@dominikbrazdil Yes, the concern was raised by @jbartece and it will definitely help.
@dominikbrazdil AFAIK DA currently needs only list of versions and in the future it will need also the artifact quality, so I would add it right away. The result would be just a map version -> quality.  @janinko Correct?

Reply:
This endpoint should be optimized as much as possible. Using the generic endpoint with RSQL is not a good option. It will be heavily used. 
I second Honza's point, if it needs to return the whole artifact. ArtifactRef should be sufficient or possibly we could even create a new DTO entity.

Reply:
I still think that RSQL parsing is not a performance issue but with Honza's suggestion, there is need for a new endpoint anyway.

Reply:
It is definitely worse than without RSQL, so there is no point in parsing the string 100 000+ times a day.